server/src/main.rs

```diff
@@ -26,31 +76,131 @@ impl SvcTelemetryRpc for SvcTelemetryImpl {
     }
 }
 
-///Main entry point: starts gRPC Server on specified address and port
-#[tokio::main]
-async fn main() -> Result<(), Box<dyn std::error::Error>> {
+/// Responds a NOT_FOUND status and error string
+///
+/// # Arguments
+///
+/// # Examples
+///
+/// ```
+/// let app = Router::new()
+///         .fallback(not_found.into_service());
+/// ```
+async fn not_found(uri: axum::http::Uri) -> impl IntoResponse {
+    (
+        axum::http::StatusCode::NOT_FOUND,
+        format!("No route {}", uri),
+    )
+}
+
+/// Tokio signal handler that will wait for a user to press CTRL+C.
+/// We use this in our hyper `Server` method `with_graceful_shutdown`.
+///
+/// # Arguments
+///
+/// # Examples
+///
+/// ```
+/// Server::bind(&"0.0.0.0:8000".parse().unwrap())
+/// .serve(app.into_make_service())
+/// .with_graceful_shutdown(shutdown_signal())
+/// .await
+/// .unwrap();
+/// ```
+async fn shutdown_signal(server: &str) {
+    tokio::signal::ctrl_c()
+        .await
+        .expect("expect tokio signal ctrl-c");
+    warn!("({}) shutdown signal", server);
+}
+
+/// Starts the grpc server for this microservice
+async fn grpc_server() {
     // GRPC Server
     let grpc_port = std::env::var("DOCKER_PORT_GRPC")
         .unwrap_or_else(|_| "50051".to_string())
         .parse::<u16>()
         .unwrap_or(50051);
 
-    let full_grpc_addr = format!("[::]:{}", grpc_port).parse()?;
-
-    let (mut health_reporter, health_service) = tonic_health::server::health_reporter();
+    let full_grpc_addr = format!("[::]:{}", grpc_port).parse().unwrap();
     let imp = SvcTelemetryImpl::default();
+    let (mut health_reporter, health_service) = tonic_health::server::health_reporter();
     health_reporter
         .set_serving::<SvcTelemetryRpcServer<SvcTelemetryImpl>>()
         .await;
 
     //start server
-    println!("Starting gRPC server at: {}", full_grpc_addr);
+    info!("(grpc) hosted at {}", full_grpc_addr);
     Server::builder()
         .add_service(health_service)
         .add_service(SvcTelemetryRpcServer::new(imp))
         .serve(full_grpc_addr)
-        .await?;
-    println!("gRPC server running at: {}", full_grpc_addr);
+        .await
+        .unwrap();
+}
+
+/// Starts the REST API server for this microservice
+pub async fn rest_server(grpc_clients: GrpcClients) {
+    let rest_port = std::env::var("DOCKER_PORT_REST")
+        .unwrap_or_else(|_| "8000".to_string())
+        .parse::<u16>()
+        .unwrap_or(8000);
+
+    let app = Router::new()
+        .fallback(not_found.into_service())
+        .route("/tlm/adsb", routing::post(rest_api::adsb))
```

Comment:
should we spell out telemetry for readability?

Reply:
https://github.com/Arrow-air/svc-telemetry/pull/5/commits/26dbe0229d09340013fc1986e94dade4d3a69c15